lib/dockerregistry/inventory.go

```diff
@@ -1434,19 +1439,71 @@ func mkPopulateRequestsForPromotionEdges(
 	}
 }
 
-func (sc *SyncContext) getPromotionEdgesFromManifests(
-	mfests []Manifest) map[PromotionEdge]interface{} {
+// GetPromotionEdgesFromManifests generates all "edges" that we need to promote.
+func (sc *SyncContext) GetPromotionEdgesFromManifests(
+	mfests []Manifest,
+	readRepos bool,
+) map[PromotionEdge]interface{} {
 
 	edges := ToPromotionEdges(mfests)
+
+	if readRepos {
+		regs := getRegistriesToRead(edges)
+		for _, reg := range regs {
+			fmt.Println("reading this reg:", reg)
```

Comment:
Is this leftover debug code?  klog maybe?

Reply:
Oops, yes, I'll use klog.